pkgs/development/interpreters/python/hooks/python-namespaces-hook.sh

```diff
@@ -1,20 +1,27 @@
 # Clean up __init__.py's found in namespace directories
+# shellcheck shell=bash
+
 echo "Sourcing python-namespaces-hook"
 
 pythonNamespacesHook() {
     echo "Executing pythonNamespacesHook"
 
-    for namespace in ${pythonNamespaces[@]}; do
+    # Python namespaces names are Python identifiers, which must not contain spaces.
+    # See https://docs.python.org/3/reference/lexical_analysis.html
+    # shellcheck disable=SC2048
+    for namespace in ${pythonNamespaces[*]-}; do
         echo "Enforcing PEP420 namespace: ${namespace}"
 
         # split namespace into segments. "azure.mgmt" -> "azure mgmt"
-        IFS='.' read -ra pathSegments <<<$namespace
-        constructedPath=$out/@pythonSitePackages@
+        # shellcheck disable=SC2206
+        local -a pathSegments=(${namespace//./ })
```

Comment:
I know that Python namespace names are restricted enough that this is fine, but I’m not in love with moving from things like `read` to these more interpolation‐y solutions. Was ShellCheck unhappy with the previous state?

Reply:
Oops! I cannot recall why I made this change. The `read` one looks better. I'll change it back.